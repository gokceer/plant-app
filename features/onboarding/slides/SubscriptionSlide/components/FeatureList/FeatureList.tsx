import React from "react";
import { ScrollView, StyleSheet, View } from "react-native";
import FeatureBox from "../FeatureBox/FeatureBox";

const features = [
  {
    icon: require("../../../../../../assets/icons/scanIcon.png"),
    title: "Unlimited",
    subtitle: "Plant Identify",
  },
  {
    icon: require("../../../../../../assets/icons/fastIcon.png"),
    title: "Faster",
    subtitle: "Process",
  },
];

const FeatureList = () => {
  return (
    <View style={styles.container}>
      <ScrollView
        horizontal
        showsHorizontalScrollIndicator={false}
        bounces={false}
      >
        {features.map((feature, index) => (
          <FeatureBox
            key={index}
            icon={feature.icon}
            title={feature.title}
            subtitle={feature.subtitle}
          />
        ))}
      </ScrollView>
    </View>
  );
};

export default FeatureList;

const styles = StyleSheet.create({
  container: { height: 150, paddingTop: 20 },
});
