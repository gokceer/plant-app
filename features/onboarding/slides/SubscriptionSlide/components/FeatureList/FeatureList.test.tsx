import { render } from "@testing-library/react-native";
import React from "react";
import FeatureList from "./FeatureList";

describe("FeatureList", () => {
  it("renders correctly and matches snapshot", () => {
    const { toJSON } = render(<FeatureList />);
    expect(toJSON()).toMatchSnapshot();
  });
});
