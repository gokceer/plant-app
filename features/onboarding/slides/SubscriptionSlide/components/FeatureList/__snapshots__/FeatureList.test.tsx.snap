// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeatureList renders correctly and matches snapshot 1`] = `
<View
  style={
    {
      "height": 150,
      "paddingTop": 20,
    }
  }
>
  <RCTScrollView
    bounces={false}
    horizontal={true}
    showsHorizontalScrollIndicator={false}
  >
    <View>
      <View
        style={
          {
            "backgroundColor": "#FFFFFF14",
            "borderRadius": 12,
            "height": 130,
            "marginRight": 8,
            "padding": 16,
            "width": 156,
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../assets/icons/scanIcon.png",
            }
          }
          style={
            {
              "height": 36,
              "width": 36,
            }
          }
        />
        <Text
          style={
            {
              "color": "#fff",
              "fontFamily": "RubikMedium",
              "fontSize": 20,
              "paddingTop": 16,
            }
          }
        >
          Unlimited
        </Text>
        <Text
          style={
            {
              "color": "#FFFFFFB2",
              "fontFamily": "RubikRegular",
              "fontSize": 12,
              "paddingTop": 4,
            }
          }
        >
          Plant Identify
        </Text>
      </View>
      <View
        style={
          {
            "backgroundColor": "#FFFFFF14",
            "borderRadius": 12,
            "height": 130,
            "marginRight": 8,
            "padding": 16,
            "width": 156,
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../assets/icons/fastIcon.png",
            }
          }
          style={
            {
              "height": 36,
              "width": 36,
            }
          }
        />
        <Text
          style={
            {
              "color": "#fff",
              "fontFamily": "RubikMedium",
              "fontSize": 20,
              "paddingTop": 16,
            }
          }
        >
          Faster
        </Text>
        <Text
          style={
            {
              "color": "#FFFFFFB2",
              "fontFamily": "RubikRegular",
              "fontSize": 12,
              "paddingTop": 4,
            }
          }
        >
          Process
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
